devito/types/grid.py

```diff
@@ -211,22 +226,35 @@ def spacing(self):
         spacing = (np.array(self.extent) / (np.array(self.shape) - 1)).astype(self.dtype)
         return as_tuple(spacing)
 
-    @property
+    @cached_property
     def spacing_symbols(self):
-        """Symbols representing the grid spacing in each SpaceDimension"""
-        return as_tuple(d.spacing for d in self.dimensions)
+        """Symbols representing the grid spacing in each SpaceDimension."""
+        return as_tuple(d.root.spacing for d in self.dimensions)
 
-    @property
+    @cached_property
     def spacing_map(self):
         """Map between spacing symbols and their values for each SpaceDimension."""
-        return dict(zip(self.spacing_symbols, self.spacing))
+        mapper = {}
+        for d, s in zip(self.dimensions, self.spacing):
+            # Special case subsampling: `Grid.dimensions` -> (xb, yb, zb)` where
+            # `xb, yb, zb` are ConditionalDimensions whose parents are SpaceDimensions
+            if d.is_Conditional:
+                try:
+                    a, b = d.spacing.args
+                    mapper[b] = s/self.dtype(a)
+                    continue
+                except (AttributeError, ValueError):
+                    pass
```

Comment:
Don't all ConditionalDimension have a `factor`, i.e `mapper[d.parent.spacing] = s/d.factor` should always be valid  (it's 1 by default for non=subsampled one` instead of the try/except.

Reply:
yeah, but `factor` isn't always used by `ConditionalDimension` (it could be None). Anyway, you may have a point, I could assume `factor!=None` and otherwise we raise the exception

Reply:
No it can't be none, if `d._factor` is `None` then `d.factor` returns 1

Reply:
https://github.com/devitocodes/devito/blob/d1b601b4670ad758d22738f33570d0d61afe646e/devito/types/dimension.py#L769

Reply:
ah! good catch